corehq/apps/hqmedia/tasks.py

```diff
@@ -149,81 +137,126 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
             fpath += '-targeted'
     else:
         dummy, fpath = tempfile.mkstemp()
+    return fpath
+
+
+def _ensure_all_media_files_in_ccz(fpath, errors):
+    with open(fpath, 'rb') as tmp:
+        with zipfile.ZipFile(tmp, "r") as z:
+            media_suites = [f for f in z.namelist() if re.search(r'\bmedia_suite.xml\b', f)]
+            if len(media_suites) != 1:
+                message = _('Could not identify media_suite.xml in CCZ')
+                errors.append(message)
+            else:
+                with z.open(media_suites[0]) as media_suite:
+                    from corehq.apps.app_manager.xform import parse_xml
+                    parsed = parse_xml(media_suite.read())
+                    resources = {node.text for node in
+                                 parsed.findall("media/resource/location[@authority='local']")}
+                    names = z.namelist()
+                    missing = [r for r in resources if re.sub(r'^\.\/', '', r) not in names]
+                    errors += [_('Media file missing from CCZ: {}').format(r) for r in missing]
+
+
+def _build_ccz_files(build, build_profile_id, include_multimedia_files, include_index_files,
+                     download_id, compress_zip, filename, download_targeted_version):
+    from corehq.apps.hqmedia.views import iter_app_files
+    files, errors, file_count = iter_app_files(
+        build, include_multimedia_files, include_index_files, build_profile_id,
+        download_targeted_version=download_targeted_version,
+    )
+
+    if toggles.CAUTIOUS_MULTIMEDIA.enabled(build.domain):
+        manifest = json.dumps({
+            'include_multimedia_files': include_multimedia_files,
+            'include_index_files': include_index_files,
+            'download_id': download_id,
+            'build_profile_id': build_profile_id,
+            'compress_zip': compress_zip,
+            'filename': filename,
+            'download_targeted_version': download_targeted_version,
+            'app': build.to_json(),
+        }, indent=4)
+        files = itertools.chain(files, [('manifest.json', manifest)])
+    return files, errors, file_count
+
+
+def _zip_ccs_files(fpath, files, current_progress, file_progress, file_count, compression, track_progress):
+    with open(fpath, 'wb') as tmp:
+        with zipfile.ZipFile(tmp, "w") as z:
+            for path, data in files:
+                # don't compress multimedia files
+                extension = os.path.splitext(path)[1]
+                file_compression = zipfile.ZIP_STORED if extension in MULTIMEDIA_EXTENSIONS else compression
+                z.writestr(path, data, file_compression)
+                current_progress += file_progress / file_count
+                if track_progress:
+                    DownloadBase.set_progress(build_application_zip, current_progress, 100)
+
+
+def create_ccz_files(build, build_profile_id, include_multimedia_files=True, include_index_files=True,
```

Comment:
Why is this `files`, plural? Looks like it still just creates one.

Reply:
i saw this as create files for the ccz and those are multiple.
do you suggest just `create_ccz` ?

Reply:
How about `create_files_for_ccz`?